src/ServiceProviderCompilationPass.php

```diff
@@ -135,11 +135,12 @@ private function getServiceDefinitionFromCallable($serviceName, $serviceProvider
             // TODO: plug the definition-interop converter here!
         }*/
         $factoryDefinition = new Definition('Class'); // TODO: in PHP7, we can get the return type of the function!
+        $factoryDefinition->setPublic(true);
```

Comment:
With more recent Sf versions, definitions have to be public (at least that's what I believe). If you think I missed something, please tell me. 😉

Reply:
I guess it really depends on how you use those services. container-interop/service-provider does not have the notion of public/private loke Symfony does. If your definitions are private, it only means you must refer to those definitions in a constructor, and you won't be able to do something like `$container->get('MyDefinition')` (which is actually rather a good thing).

Of course, there are times where being able to call `$container->get('MyDefinition')` is useful (especially when you want to do lazy loading). The best solution here would maybe be to add a flag to the ServiceProviderBridgeBundle constructor to be able to specify if definitions are public or not.

Reply:
Meanwhile, I'm :+1: making this public by default.

Reply:
I completely agree but there are cases where constructor dependency injection cannot happen (I can give an example if needed), and the fact is that service providers don't have a notion of private/public services; this is only in Symfony (maybe in other frameworks as well but that's not my point).

So I think bothering the user with a flag in the bundle is not really needed, but maybe we'd need an input from someone else. 😊

Reply:
Ok, let's merge this then!